lib/failure.sh

```diff
@@ -27,6 +27,20 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_dot_heroku_node() {
+  if [ -f "${1:-}/.heroku/node" ]; then
```

Comment:
if `$1` is empty, it'll check `/.heroku/node`; also, "`:-`" will be used even if the arg is empty, which may be intentional.

Maybe use `"${1-}${1:+/}.heroku/node"` instead. That should yield the following expansions:

- `fail_dot_heroku_node BASEDIR` => `BASEDIR/.heroku/node`
- `fail_dot_heroku_node` => `.heroku/node`
- `fail_dot_heroku_node ""` => `.heroku/node`